code/game/objects/items/v8_engine.dm

```diff
@@ -0,0 +1,99 @@
+
+/obj/item/v8_engine/attack_self(mob/user, modifiers)
+	. = ..()
+	if (!COOLDOWN_FINISHED(src, engine_sound_cooldown))
+		return
+	playsound(src, 'sound/items/car_engine_start.ogg', vol = 75, vary = FALSE, extrarange = 3)
+	Shake(7, 7, ENGINE_COOLDOWN)
+	to_chat(user, span_notice("Darn thing... it's too old to keep on without retrofitting it! Without modifications, it works like it's junk."))
+	COOLDOWN_START(src, engine_sound_cooldown, ENGINE_COOLDOWN)
+
+/obj/item/v8_engine/examine_more(mob/user)
+	. = ..()
+	var/datum/crafting_recipe/house_edge/edge
+	to_chat(user, span_notice("You peer at the label on the side, reading about some unique modifications that could be made to the engine..."))
+	if(do_after(user, 15 SECONDS, src))
+		if(!user.mind)
+			return
+		user.mind.teach_crafting_recipe(edge)
+		to_chat(user, span_notice("You learned how to make the House Edge."))
+
+/obj/item/house_edge
+	name = "House Edge"
+	desc = "Dangerous. Loud. Sleek. It has a built in roulette wheel. This thing could easily rip your arm off if you're not careful."
+	icon = 'icons/obj/weapons/items_and_weapons.dmi'
+	icon_state = "house_edge0"
+	w_class = WEIGHT_CLASS_HUGE
+	force = 12
+	throwforce = 10
+	throw_range = 5
+	throw_speed = 1
+	hitsound = 'sound/items/car_engine_start.ogg'
+	/// The number of charges the house edge has accrued through 2-handed hits, to charge a more powerful charge attack.
+	var/fire_charges = 0
+	///Sound played when wielded.
+	var/active_hitsound = 'sound/items/house_edge_hit.ogg'
+	///Datum that tracks weapon dashing for the fire_charge system
+	var/datum/action/innate/dash/charge
+	COOLDOWN_DECLARE(fire_charge_cooldown)
+
+/obj/item/house_edge/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded = 12, force_wielded = 22, attacksound = active_hitsound)
+	RegisterSignal(src, list(COMSIG_ITEM_DROPPED, COMSIG_MOVABLE_PRE_THROW, COMSIG_ITEM_ATTACK_SELF), .proc/reset_charges)
+
+/obj/item/house_edge/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!ismob(target))
+		return
+	if(HAS_TRAIT(src, TRAIT_WIELDED))
+		//Add a fire charge to a max of 3, updates icon_state.
+		fire_charges = clamp((fire_charges + 1), 0, 3)
+		icon_state = "house_edge[fire_charges]"
+		COOLDOWN_RESET(src, fire_charge_cooldown)
+	else
+		//Lose a fire charge to a min of 0, updates icon_state.
+		fire_charges = clamp((fire_charges - 1), 0, 3)
+		icon_state = "house_edge[fire_charges]"
+		do_sparks(0, TRUE, src)
+
+/obj/item/house_edge/pre_attack_secondary(atom/target, mob/living/user, params)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, fire_charge_cooldown))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(fire_charges <= 0)
+		src.balloon_alert(user, "No fire charges")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	user.throw_at(target = get_turf(target), range = 2 * fire_charges, speed = 5, thrower = user, spin = FALSE, force = 6 * fire_charges, gentle = FALSE, quickstart = TRUE)
```

Comment:
what are you doing with the force argument? its not damage, that is how strong the move force of the thrown object is, can the impact move the object it was thrown at

Reply:
Should also set gentle to TRUE so you don't wipe out hitting literally anything.

Reply:
I set gentle to FALSE intentionally to play into the risk reward of the weapon. You have a way to dash, but the way forward needs to be clear so that you don't crash the sword straight into the ground.

Force arg is on point however, I thought that was like thrown force of the tackle.

Reply:
I can tell ya from using dash boots that it's not nearly as intuitive as you'd hope, because you usually can't always tell when you'll smash into an object. Dense objects are all over the shop that don't fully fill the screen. You can also accidentally intercept someone who fidgets about, which a lot of players just kind of do, so you can't really close on anyone who isn't more or less typing or using a console.